modules/openideal_rest/src/Plugin/RulesAction/WebhookAction.php

```diff
@@ -0,0 +1,166 @@
+    $plugin_id,
+    $plugin_definition,
+    WebhooksService $webhookService,
+    EntityTypeManagerInterface $entityTypeManager,
+    SerializerInterface $serializer
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->webhookService = $webhookService;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->serializer = $serializer;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('webhooks.service'),
+      $container->get('entity_type.manager'),
+      $container->get('serializer')
+    );
+  }
+
+  /**
+   * Dispatch webhooks event.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Entity from fetch data.
+   * @param string $event
+   *   Event id.
+   */
+  protected function doExecute(EntityInterface $entity, $event) {
+    // Remove default part of id.
+    $id = preg_replace('/[a-z0-9_]+:/', '', $this->getPluginId());
+
+    switch (strtolower($id)) {
+      case 'slack':
+        $this->reactOnSlack($entity, $event);
+        return;
+
+      default:
+        $this->reactOnDefault($entity, $event);
+
+    }
+  }
+
+  /**
+   * Send slack webhooks.
+   *
+   * @param \Drupal\message\MessageInterface $entity
+   *   Entity.
+   * @param string $event
+   *   Event.
+   */
+  protected function reactOnSlack(MessageInterface $entity, $event) {
+    $webhook_configs = $this->getWebConfigByEventAndPlugin($event, 'slack');
+
+    $template_id = $entity->getTemplate()->id();
+    // Only user_joined event don't have message partial field for
+    // email because we don't have ability to follow the user.
+    $delta = $template_id == 'user_joined' ? 1 : 3;
+    $delta = $template_id == 'user_mention' ? 4 : $delta;
+
+    // @todo Implement logic to identify message texts inside partial fields
+    //   or create another field.
+    $text = $entity->getText(NULL, $delta);
+    $text = Json::decode(reset($text));
+    foreach ($webhook_configs as $webhook_config) {
+      $webhook = new Webhook(
+        ['event' => $event] + $text,
+        [],
+        $event,
+        $webhook_config->getContentType()
+      );
+      $this->webhookService->send($webhook_config, $webhook);
+    }
+  }
+
+  /**
+   * Default handler.
+   */
+  protected function reactOnDefault($entity, $event) {
+    // @todo Implement.
+  }
+
+  /**
+   * Load webconfig by event and plugin.
+   *
+   * @param string $event
+   *   Event type.
+   * @param string $plugin
+   *   Plugin type.
+   *
+   * @return \Drupal\webhooks\Entity\WebhookConfig[]
+   *   Webconfig entities.
+   */
+  protected function getWebConfigByEventAndPlugin($event, $plugin) {
+    $storage = $this->entityTypeManager->getStorage('webhook_config');
+    $query = $storage->getQuery()
```

Comment:
I assume access check isn't necessary here, right?
Let's add `accessCheck(FALSE)` in that case.

Reply:
Done